vizro-core/tests/unit/vizro/models/_components/test_aggrid.py

```diff
@@ -0,0 +1,156 @@
+"""Unit tests for vizro.models.AgGrid."""
+
+import dash_ag_grid as dag
+import pytest
+from asserts import assert_component_equal
+from dash import dcc, html
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+import vizro.plotly.express as px
+from vizro.managers import data_manager
+from vizro.models._action._action import Action
+from vizro.tables import dash_ag_grid
+
+
+@pytest.fixture
+def dash_ag_grid_with_arguments():
+    return dash_ag_grid(data_frame=px.data.gapminder(), defaultColDef={"resizable": False, "sortable": False})
+
+
+@pytest.fixture
+def dash_ag_grid_with_str_dataframe():
+    return dash_ag_grid(data_frame="gapminder")
+
+
+class TestDunderMethodsAgGrid:
```

Comment:
I think all these `test_create` things probably belong in a class that's not `TestDunderMethods` but instead `TestAgGridInstantation` like we do for other models.